### PR TITLE
Spline #175 Change dependencies of spark to the provided scope 

### PR DESCRIPTION
**Pull request by leborgneromain**
+cleanup unused dependencies

### CODE REVIEW

**Comment by vackosar** [comment]
good spot!

File: core/pom.xml [hunk 1]
```diff
@@ -66,6 +58,16 @@
             <groupId>org.scalaz</groupId>
             <artifactId>scalaz-core_${scala.compat.version}</artifactId>
         </dependency>
+        <dependency>
```
Comment (vackosar) [inline core/pom.xml]: Do you need these at all? Aren't they inherited from spline-spark-api?

Comment (leborgneromain) [reply core/pom.xml]: yes for the sample project at least

Comment (vackosar) [reply core/pom.xml]: What I mean is that ```spline-core``` inherits ```spark-core_${scala.compat.version}``` and ```spark-sql_${scala.compat.version}``` from ```spline-core-spark-adapter-api```. So you shouldn't need to specify them again in ```spline-core```.

Comment (vackosar) [reply core/pom.xml]: sry actually provided scope is not transitive

**Pull request closed without merging**